Public/Src/FrontEnd/MsBuild/PipConstructor.cs

```diff
@@ -330,11 +332,14 @@ public bool TrySchedulePipForFile(ProjectWithPredictions project, QualifierId qu
                     continue;
                 }
 
-                // Nuget restore generates nuget.g.props files that have absolute paths embedded. This blocks shared cache from working when
-                // the project file (and therefore the corresponding .pkgrefgen folder) is placed on machine-dependent folders
-                // Even though untracking this file is not completely safe from a caching perspective, in practice this should be harmless
+                // Nuget restore does not produce deterministic files under the project .pkgrefgen folder, and many times they have absolute paths embedded. 
+                // This blocks shared cache from working when the project file (and therefore the corresponding .pkgrefgen folder) is placed on machine-dependent folders
+                // Even though untracking these generated files is not completely safe from a caching perspective, in practice this should be harmless since the generation
+                // is controlled by files (e.g. project.assents.json or global.json) that are declared as inputs.
                 // In this case, we not only skip declaring it as an input, but we also untrack it as well
-                if (buildInput.GetName(PathTable).ToString(PathTable.StringTable).EndsWith("nuget.g.props", StringComparison.OrdinalIgnoreCase))
+                if (buildInput.GetParent(PathTable) is AbsolutePath parent && 

```

Comment:
I assume `nuget.g.props` is under the `.pkgrefgen` directory?

Reply:
Yes